Framework/Core/src/ExternalFairMQDeviceProxy.cxx

```diff
@@ -241,49 +241,63 @@ InjectorFunction dplModelAdaptor(std::vector<OutputSpec> const& filterSpecs, boo
-      bool indexDone = false;
+      size_t finalBlockIndex = 0;
+      std::string channelName = "";
+
       for (auto const& spec : filterSpecs) {
         // filter on the specified OutputSpecs, the default value is a ConcreteDataTypeMatcher with origin and description 'any'
         if (DataSpecUtils::match(spec, OutputSpec{{header::gDataOriginAny, header::gDataDescriptionAny}}) ||
             DataSpecUtils::match(spec, query)) {
-          auto channelName = channelRetriever(query, dph->startTime);
+          channelName = channelRetriever(query, dph->startTime);
           if (channelName.empty()) {
             LOG(warning) << "can not find matching channel, not able to adopt " << DataSpecUtils::describe(query);
-            break;
-          }
-          // the checks for consistency of split payload parts are of informative nature
-          // forwarding happens independently
-          if (dh->splitPayloadParts > 1 && dh->splitPayloadParts != std::numeric_limits<decltype(dh->splitPayloadParts)>::max()) {
-            if (lastSplitPartIndex == -1 && dh->splitPayloadIndex != 0) {
-              LOG(warning) << "wrong split part index, expecting the first of " << dh->splitPayloadParts << " part(s)";
-            } else if (dh->splitPayloadIndex != lastSplitPartIndex + 1) {
-              LOG(warning) << "unordered split parts, expecting part " << lastSplitPartIndex + 1 << ", got " << dh->splitPayloadIndex
-                           << " of " << dh->splitPayloadParts;
-            } else if (channelNameForSplitParts.empty() == false && channelName != channelNameForSplitParts) {
-              LOG(error) << "inconsistent channel for split part " << dh->splitPayloadIndex
-                         << ", matching " << channelName << ", expecting " << channelNameForSplitParts;
-            }
-            lastSplitPartIndex = dh->splitPayloadIndex;
-            channelNameForSplitParts = channelName;
-            if (lastSplitPartIndex + 1 == dh->splitPayloadParts) {
-              lastSplitPartIndex = -1;
-              channelNameForSplitParts = "";
-            }
-          } else if (lastSplitPartIndex != -1) {
-            LOG(warning) << "found incomplete or unordered split parts, expecting part " << lastSplitPartIndex + 1
-                         << " but got a new data block";
           }
-          outputs[channelName].AddPart(std::move(parts.At(msgidx * 2)));
-          outputs[channelName].AddPart(std::move(parts.At(msgidx * 2 + 1)));
-          LOG(debug) << "associating part with index " << msgidx << " to channel " << channelName << " (" << outputs[channelName].Size() << ")";
-          indexDone = true;
           break;
         }
       }
-      if (indexDone == false && !DataSpecUtils::match(query, "DPL", "EOS", 0)) {
+      if (!channelName.empty()) {
```

Comment:
I would invert the logic and have a `continue` on `channelName.empty() == true`.

Reply:
ah, yeah, can revise the logic. Actually I did not want to continue in the loop because the unmatched specs are logged. But I will combine this, there is room for cleanup in the logic.